test/validation/api/timer/timer.c

```diff
@@ -1218,8 +1277,31 @@ static int worker_entrypoint(void *arg ODP_UNUSED)
 	odp_pool_t tbp = global_mem->tbp;
 	uint32_t num_timers = global_mem->timers_per_thread;
 	uint64_t min_tmo = global_mem->param.min_tmo;
+	odp_timer_capability_t capa;
+	odp_queue_param_t queue_param;
+	odp_queue_type_t queue_type = ODP_QUEUE_TYPE_PLAIN;
+	odp_thrmask_t thr_mask;
+	uint64_t sched_tmo;
+	int ret;
 
-	queue = odp_queue_create("timer_queue", NULL);
+	memset(&capa, 0, sizeof(capa));
+	ret = odp_timer_capability(ODP_CLOCK_CPU, &capa);
+	CU_ASSERT_FATAL(ret == 0);
+
+	odp_queue_param_init(&queue_param);
+	if (capa.queue_type_plain) {
+		queue_param.type = ODP_QUEUE_TYPE_PLAIN;
+		queue_type = ODP_QUEUE_TYPE_PLAIN;
+	} else if (capa.queue_type_sched) {
```

Comment:
This if-else if tests only plain queues when both are supported. So, reference implementation runs only the plain queue path of the test. The scheduled queue path should be also tested with reference implementation (to be  sure it works and will remain working in the future).

Could you add queue type parameter to timer_test_all, and run it twice in the suite (once for plain and once for sched).

Reply:
Makes sense, will split the test case.